include/imgproc/dct_hash.h

```diff
@@ -0,0 +1,82 @@
+#pragma once
+
+// Standard C++ includes
+#include <iostream>
+#include <string>
+#include <algorithm>
+
+// Standard C includes
+#include <cmath>
+
+//opencv
+#include <opencv2/opencv.hpp>
+
+
+class DCTHash {
+
+    private:
+    static unsigned int countSetBits(unsigned long long int n) { 
+        unsigned int count = 0; 
+        while (n) { 
+            count += n & 1; 
+            n >>= 1; 
+        } 
+        return count; 
+    } 
+
+    public:
+    
+    
+    //! computes the DCT hash 
+    static unsigned long long computeHash(const cv::Mat &in) {
+        
+		unsigned long long hash;
+        cv::Mat dct_mat;
+        cv::dct(in, dct_mat);
+        cv::Rect roi( 0, 0, 8, 8); 
+        cv::Mat low_dct( dct_mat, roi ); 
+
+		std::vector<float> vec;
+		for (int row = 0; row < 9; row++) {
+			for (int col = 0; col < 9; col++) {
+			vec.push_back(low_dct.at<float>(row,col));
+			}
+		}
+
+		// calculate median value 
+		std::sort(std::begin(vec), std::end(vec));
```

Comment:
Glad to see you went with the STL algo in the end :wink:. As I mentioned, you could also probably use ``std::partial_sort`` for a slight speedup, but that's not terribly important...

You could also drop ``vec`` and just operate on ``low_dct`` directly:
```c++
std::sort(low_dct.begin<float>(), low_dct.end<float>());
```
etc. I know ``vec`` is tiny but we want to avoid unnecessary heap allocations!

Reply:
Ignore this, I just noticed you're reusing ``vec``...